test/Authentication/Storage/CookieTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Carnage\JwtZendAuthTest\Authentication\Storage;
+
+use Carnage\JwtZendAuth\Authentication\Storage\Cookie;
+use Mockery\Adapter\Phpunit\MockeryTestCase;
+use Zend\Console\Request as ConsoleRequest;
+use Zend\Http\Header\Cookie as CookieHeader;
+use Zend\Http\Request;
+use Zend\Http\Response;
+
+class CookieTest extends MockeryTestCase
+{
+    /**
+     * @dataProvider provideRead
+     * @param $request
+     * @param $expected
+     */
+    public function testRead($request, $expected)
```

Comment:
Probably worth adding a test to check calling read twice in a row, given that there's logic explicit logic around it